kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/storageApi/StorageWriteApiBase.java

```diff
@@ -124,16 +133,58 @@ protected Map<Integer, String> getRowErrorMapping(Exception exception) {
         }
     }
 
+    protected boolean getAutoCreateTables() {
+        return this.autoCreateTables;
+    }
+
     /**
-     * Wait at least {@link #retryWait}, with up to an additional 1 second of random jitter.
-     * @throws InterruptedException if interrupted.
+     * Attempts to create table
+     * @param retryHandler Plain java object which handles retries on table creation
      */
-    protected void waitRandomTime() throws InterruptedException {
-        Thread.sleep(retryWait + random.nextInt(1000));
+    protected void attemptTableCreation(RetryHandler retryHandler) {
+        try {
+            if (!retryHandler.isTableCreationOrUpdateAttempted()) {
+                retryHandler.setTableCreationOrUpdateAttempted(true);
+                schemaManager.createTable(retryHandler.getTableId(), retryHandler.getRecords());
+                // Table takes time to be available for after creation
+                retryHandler.setAdditionalRetriesAndWait();
+            } else {
+                logger.info("Skipping multiple table creation attempts");
+            }
+        } catch (BigQueryException exception) {
+            throw new BigQueryStorageWriteApiConnectException(
+                    "Failed to create table " + retryHandler.getTableId(), exception);
+        }
     }
 
-    protected boolean getAutoCreateTables() {
-        return this.autoCreateTables;
+    /**
+     * Attempts to update table schema
+     * @param retryHandler Plain java object which handles retries on table schema update
+     */
+    protected void attemptSchemaUpdate(RetryHandler retryHandler) {
```

Comment:
methods attemptTableCreation and attemptSchemaUpdate can be refactored into one to pass a lambda for createTable and updateSchema

Reply:
Thanks for the suggestion, updated.